go-mode.el

```diff
@@ -1751,10 +1751,16 @@ The following extra functions are defined:
 - `godef-describe' and `godef-jump'
 - `go-coverage'
 
-If you want to automatically run `gofmt' before saving a file,
-add the following hook to your Emacs configuration:
+If you want to automatically run `gofmt' before saving a file enable the
+gofmt minor mode.  To do so you can add the following to to your Emacs
+configuration:
 
-\(add-hook 'before-save-hook #'gofmt-before-save)
+\(add-hook 'go-mode-hook #'gofmt-mode)
+
+Or if you use `use-package':
+
+\(use-package go-mode
+  :hook ((go-mode . gofmt-mode)))
```

Comment:
I included the use-package instructions because it is now included with Emacs. Let me know if you prefer to remove it